kickoff/jsonexportl10n.py

```diff
@@ -74,12 +101,33 @@ def jsonRegionsExports():
 
 def generateListPerProduct(product):
     releases = getReleases(ready=True, productFilter=product)
-    version_list = []
+    registrar = _L10nReleasesRegistrar()
     for r in releases:
-        if r._shippedAt and r.l10nChangesets != config.LEGACY_KEYWORD:
-            # Only list version which shipped with l10n content
-            version_list.append(r.name)
-    return version_list
+        registrar.addRelease(r)
+    return registrar.releases
+
+
+class _L10nReleasesRegistrar:
+    # Matches strings like "Firefox-19.0b1-build1" and stores "Firefox-19.0b"
+    BETA_REGEX = re.compile(r"([^\-]+-(\d+\.)+0b)\d+-build\d+")
+
+    def __init__(self):
+        self.releases = []
+        self._betas_already_processed = set()
+
+    def addRelease(self, release):
+        if release.isShippedWithL10n:
+            self._addAggregatedBetaIfNeeded(release)
+            self.releases.append(release.name)
+
+    def _addAggregatedBetaIfNeeded(self, release):
+        first_beta_match = self.BETA_REGEX.match(release.name)
```

Comment:
I renamed the function into `_addAggregatedBetaOnlyOnce()` in order to make the intent of this function clearer. Please tell me if you'd like more details